sql/mozfun/iap/scrub_apple_receipt/udf.sql

```diff
@@ -2,114 +2,37 @@ CREATE OR REPLACE FUNCTION iap.scrub_apple_receipt(apple_receipt ANY TYPE)
-      ),
-      stage_2 AS (
-        SELECT
-          TIMESTAMP_MILLIS(purchase_date_ms) AS start_time,
-          TIMESTAMP_MILLIS(expires_date_ms) AS end_time,
-          iap.derive_apple_subscription_interval(
-            DATETIME(TIMESTAMP_MILLIS(purchase_date_ms), "America/Los_Angeles"),
-            DATETIME(TIMESTAMP_MILLIS(expires_date_ms), "America/Los_Angeles")
-          ).*,
-        FROM
-          stage_1
-        WHERE
-          is_trial_period = "false"
-      ),
-      stage_3 AS (
-        SELECT DISTINCT
-          date,
-          IF(
-            -- TRUE when date - 1 is missing
-            ANY_VALUE(date) OVER (
-              PARTITION BY
-                `interval`,
-                interval_count
-              ORDER BY
-                UNIX_DATE(date)
-              RANGE BETWEEN
-                1 PRECEDING
-                AND 1 PRECEDING
-            ) IS NULL,
-            date,
-            NULL
-          ) AS start_date,
-          IF(
-            -- TRUE when date + 1 is missing
-            ANY_VALUE(date) OVER (
-              PARTITION BY
-                `interval`,
-                interval_count
-              ORDER BY
-                UNIX_DATE(date)
-              RANGE BETWEEN
-                1 FOLLOWING
-                AND 1 FOLLOWING
-            ) IS NULL,
-            date + 1,
-            NULL
-          ) AS end_date,
-          `interval`,
-          interval_count,
-        FROM
-          stage_2
-        CROSS JOIN
-          UNNEST(
-            GENERATE_DATE_ARRAY(
-              DATE(start_time),
-              -- don't include the last day, it will be included on the next receipt
-              DATE(end_time) - 1
-            )
-          ) AS date
-      ),
-      stage_4 AS (
-        SELECT
-          start_date,
-          -- MIN(end_date) where end_date > start_date
-          MIN(end_date) OVER (
-            PARTITION BY
-              `interval`,
-              interval_count
-            ORDER BY
-              date
-            ROWS BETWEEN
-              CURRENT ROW
-              AND UNBOUNDED FOLLOWING
-          ) AS end_date,
-          `interval`,
-          interval_count,
-        FROM
-          stage_3
-      )
       SELECT AS STRUCT
-        *
+        DATE(TIMESTAMP_MILLIS(original_purchase_date_ms)) AS start_date,
+        DATE(TIMESTAMP_MILLIS(expires_date_ms)) AS end_date,
+        TIMESTAMP_MILLIS(original_purchase_date_ms) AS start_time,
+        TIMESTAMP_MILLIS(expires_date_ms) AS end_time,
+        iap.derive_apple_subscription_interval(
+          DATETIME(TIMESTAMP_MILLIS(purchase_date_ms), "America/Los_Angeles"),
+          DATETIME(TIMESTAMP_MILLIS(expires_date_ms), "America/Los_Angeles")
+        ).*,
       FROM
-        stage_4
+        UNNEST(apple_receipt.latest_receipt_info)
       WHERE
-        start_date IS NOT NULL
+        is_trial_period = "false"
+      ORDER BY
+        expires_date_ms DESC
+      LIMIT
+        1
```

Comment:
the apple receipt [excludes old transactions](https://github.com/mozilla-services/guardian-website/blob/c9a3d9def4c2643ebd177f1d1afa9963bdf87801/server/lib/apple.ts#L124), so this must assume that the subscription is continuous from `original_purchase_date_ms` through `expires_date_ms`